src/models/encrypted-model.ts

```diff
@@ -0,0 +1,32 @@
+import { Model } from "sequelize";
+import { EncryptionClient, EncryptionContext, EncryptionSecretKeyEnum } from "utils/encryption-client";
+
+type StringValues<Obj> = {
+	[Prop in keyof Obj]: Obj[Prop] extends string ? Prop : never
+}[keyof Obj];
+
+export abstract class EncryptedModel<C extends Model, K extends keyof Pick<C, StringValues<C>>> extends Model {
+
+	abstract getEncryptContext(field: K): Promise<Record<string, string | number>>;
+
+	abstract getEncryptionSecretKey(): EncryptionSecretKeyEnum;
+
+	abstract getSecretFields(): readonly K[];
+
+	async decrypt(value: string, context?: EncryptionContext): Promise<string> {
+		return await EncryptionClient.decrypt(value, context);
```

Comment:
hi @mboudreau , shouldn't this be `async decrypt(value: K` instead of `async decrypt(value: string`